src/daemon/rpmostreed-daemon.cxx

```diff
@@ -774,6 +775,47 @@ rpmostreed_daemon_exit_now (RpmostreedDaemon *self)
   self->running = FALSE;
 }
 
+static gboolean
+idle_initiate_reboot (void *_unused)
+{
+  sd_journal_print (LOG_INFO, "Initiating reboot requested from transaction");
+
+  /* Note that we synchronously spawn this command, but the command just queues the request and returns.
+   */
+  const char *child_argv[] = { "systemctl", "reboot", NULL };
+  g_autoptr(GError) local_error = NULL;
+  if (!g_spawn_sync (NULL, (char**)child_argv, NULL, (GSpawnFlags)(G_SPAWN_CHILD_INHERITS_STDIN | G_SPAWN_SEARCH_PATH),
```

Comment:
What happens if there is a systemd inhibitor lock that is blocking shutdown? Would the `systemctl reboot` called in this function fail later when the reboot actually occurs, or error immediately and the daemon exits? 
If the former, when used in `finalize_deployment()`, would we have an unlocked staged deployment since we'd `unlink()` `_OSTREE_SYSROOT_RUNSTATE_STAGED_LOCKED` but not actually reboot and return a success message to the client even if we fail to reboot?
So it seems like both of these situations are not ideal. (I may also just have misunderstood systemd-inhibit and these bits of code entirely)

Reply:
Good point!  So...Per [this point](https://github.com/coreos/zincati/issues/245#issuecomment-844274765) I think really the only correct thing here is for e.g. zincati to not try to finalize if there are inhibitors.    I can't think of a way to solve this without adding some sort of new inhibitor API to do something more read/write lock like so that one could do "grab an exclusive lock only when the last (read) locker has dropped".

That said honestly I was just playing with this locally and I assume I'm doing something wrong, but I can't get `systemd-inhibit` to block reboots.  I'm doing `systemd-inhibit sleep 1h` in one shell, and `systemctl reboot` in another instantly reboots.

Reply:
(OK some investigation shows I need to use `systemctl reboot --check-inhibitors=yes` which...seems like a bug, the man page claims `auto` will do that on a tty, but that's not what I'm seeing)

Reply:
Hmm, so the `--check-inhibitors` option was made available in a [recent-ish commit](https://github.com/systemd/systemd/pull/18210), and is not in FCOS yet; but even so, reboots should be inhibited successfully if called on a tty. I tried locally with `systemd-inhibit sleep 1h &`, then if I do `systemctl reboot`, it is indeed inhibited.
```
systemctl poweroff
Operation inhibited by "sleep 1h" (PID 2653 "systemd-inhibit", user root), reason is "Unknown reason".
Please retry operation after closing inhibitors and logging out other users.
Alternatively, ignore inhibitors and users with 'systemctl poweroff -i'.
```
However, another issue I see right now regarding `systemd-inhibit` is that I haven't been able to get my inhibitor locks to inhibit rpm-ostree-driven reboots, even if I set the lock as `root`. I assume this is because the rpm-ostree daemon is privileged and the `systemctl reboot` called by the daemon is non-interactive, and so in e.g. `systemd 246` (before https://github.com/systemd/systemd/pull/18210), any inhibitors are ignored by rpm-ostree. So IIUC, systemd-inhibitors won't really work with rpm-ostree today, at least not until `systemd 248`, after which we can add the `--check-inhibitors=yes` option?

Reply:
OK yeah if you follow issue links there's a *huge* amount of debate about this.